src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -1207,17 +1382,46 @@ - (void)subscribeToAttributesWithEndpointID:(NSNumber * _Nullable)endpointID
                        }
                    };
 
+                   auto onResubscriptionScheduledCb = [resubscriptionScheduled](NSError * error, NSNumber * resubscriptionDelay) {
+                       if (resubscriptionScheduled) {
+                           resubscriptionScheduled(error, resubscriptionDelay);
+                       }
+                   };
+
                    MTRReadClientContainer * container = [[MTRReadClientContainer alloc] init];
                    container.deviceID = self.nodeID;
-                   container.pathParams = Platform::New<app::AttributePathParams>();
-                   if (endpointID) {
-                       container.pathParams->mEndpointId = static_cast<chip::EndpointId>([endpointID unsignedShortValue]);
-                   }
-                   if (clusterID) {
-                       container.pathParams->mClusterId = static_cast<chip::ClusterId>([clusterID unsignedLongValue]);
+
+                   size_t attributePathSize = 0;
+                   if (attributes != nil) {
+                       container.pathParams = static_cast<AttributePathParams *>(
+                           Platform::MemoryCalloc([attributes count], sizeof(AttributePathParams)));
+                       if (container.pathParams == nullptr) {
+                           if (reportHandler) {
```

Comment:
reportHandler can't be nil here, so no need to null-check that.

Reply:
Removed.